pkg/hoist/hoist_launchable.go

```diff
@@ -322,10 +322,16 @@ func (hl *Launchable) Install(verifier auth.ArtifactVerifier) error {
 	return err
 }
 
-// The version of the artifact is currently derived from the location, using
-// the naming scheme <the-app>_<unique-version-string>.tar.gz
+func (hl *Launchable) tempFilePath() string {
+	return fmt.Sprintf("%s_%s.tar.gz", hl.Id, hl.Version())
```

Comment:
`Version()` is everything in the file name except the extension, so this is going to produce temp file paths like `foo_foo_1234`. Is that fine?